notebooks/registry.ipynb

```diff
@@ -0,0 +1,615 @@
+      "cell_type": "code",
+      "source": [
+        "from hive.utils.registry import registry\n",
+        "registry.register(name = 'LearningAgent', \n",
+        "                  constructor = LearningAgent, \n",
+        "                  type = Agent)"
+      ],
+      "metadata": {
+        "id": "slMh_1bR-s1a"
+      },
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "More than one agent can be registered at once using the `register_all` method. Consider registering three environments, `LearningAgentV1`, `LearningAgentV2`, and `LearningAgentV3` in the RLHive registry."
+      ],
+      "metadata": {
+        "id": "kdcN7NZlT6dD"
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "registry.register_all(\n",
+        "    Agent,\n",
+        "    {\n",
+        "        \"LearningAgentV1\": LearningAgentV1,\n",
+        "        \"LearningAgentV2\": LearningAgentV2,\n",
+        "        \"LearningAgentV3\": LearningAgentV3,\n",
+        "    },\n",
+        ")"
+      ],
+      "metadata": {
+        "id": "WUDq9CoJMYXf"
+      },
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "### Registering a Logger"
+      ],
+      "metadata": {
+        "id": "Ku9Jz5EKbmRE"
+      }
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "Consider registering a custom environment class named `CustomLogger` (which inherits `Logger` class) in the RLHive registry. "
+      ],
+      "metadata": {
+        "id": "ureIVBKcbtZJ"
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "from hive.utils.loggers import Logger\n",
+        "\n",
+        "class CustomLogger(Logger):\n",
+        "    def __init__(self):\n",
+        "        pass\n",
+        "    def update_step(self, timescale):\n",
+        "        pass"
+      ],
+      "metadata": {
+        "id": "70ys0SvobnkD"
+      },
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "from hive.utils.registry import registry\n",
+        "registry.register(name = 'CustomLogger', \n",
+        "                  constructor = CustomLogger, \n",
+        "                  type = Logger)"
+      ],
+      "metadata": {
+        "id": "otDqYlUWcDkn"
+      },
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "More than one logger can be registered at once using the `register_all` method. Consider registering three environments, `CustomLoggerV1`, `CustomLoggerV2`, and `CustomLoggerV3` in the RLHive registry."
+      ],
+      "metadata": {
+        "id": "A8ebt6CvdYLX"
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "registry.register_all(\n",
+        "    Agent,\n",
+        "    {\n",
```

Comment:
I think there is a typo here the base class should be Logger instead of Agent in this case, let me know if I am wrong.